### PR TITLE
js feat: Support attachments in traceable

### PR DESCRIPTION
**Pull request by hinthornw**
Also add warning & handling of invalid attachments.

Proposed API:

getAttachments(args) => [attachments, otherArgs];

Returning the filtered args means that you're less likely to accidentally double-log attachments in inputs.

### CODE REVIEW

File: js/src/tests/traceable.int.test.ts [hunk 1]
```diff
@@ -651,9 +651,105 @@ test.concurrent(
...
-    await waitUntilRunFound(langchainClient, runId);
-    const storedRun3 = await langchainClient.readRun(runId);
+    await waitUntilRunFound(langsmithClient, runId);
+    const storedRun3 = await langsmithClient.readRun(runId);
     expect(storedRun3.id).toEqual(runId);
     expect(storedRun3.status).toEqual("error");
   }
 );
+
+test.only("Test upload attachments and process inputs.", async () => {
```
Comment (dqbd) [inline js/src/tests/traceable.int.test.ts:661]: Remove the scoped test

File: js/src/schemas.ts [hunk 2]
```diff
@@ -587,3 +593,5 @@ export type UsageMetadata = {
    */
   output_token_details?: OutputTokenDetails;
 };
+
+export type Attachments = Record<string, [string, Uint8Array]>;
```
Comment (dqbd) [inline js/src/schemas.ts:597]: _question_: do we want to support ArrayBuffer as well? Might save a unnecessary array copy

File: js/src/tests/traceable.int.test.ts (hunk 1 above)
Comment (hinthornw) [reply js/src/tests/traceable.int.test.ts:661]: oops

File: js/src/schemas.ts (hunk 2 above)
Comment (hinthornw) [reply js/src/schemas.ts:597]: Oh - I think that makes sense? Will have to look into it

Comment (hinthornw) [reply js/src/schemas.ts]: Added!

File: js/src/traceable.ts [hunk 3]
```diff
@@ -309,6 +338,15 @@ export function traceable<Func extends (...args: any[]) => any>(
     argsConfigPath?: [number] | [number, string];
     __finalTracedIteratorKey?: string;
 
+    /**
+     * Specify which inputs should be treated as attachments and return the remaining arguments.
+     * @param args Arguments of the traced function
+     * @returns A tuple containing the Attachments object and the remaining arguments
+     */
+    getAttachments?: (
```
Comment (agola11) [inline js/src/traceable.ts:346]: calling this `getAttachments` makes it seem like this should return _just_ the attachments? `separateAttachments` or something similar would be better IMO

**Review by agola11 — APPROVED** [review]

**Pull request merged**